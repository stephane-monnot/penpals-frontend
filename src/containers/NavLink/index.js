import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const NavLink = (props) => (
  <li>
    <Link {...props} />
  </li>
);

NavLink.propTypes = {
  to: PropTypes.string.isRequired
};

export default NavLink;
