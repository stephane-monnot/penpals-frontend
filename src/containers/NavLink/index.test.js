import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';
import { Link } from 'react-router-dom';
import NavLink from './';

describe(NavLink, () => {
  const component = mount(
    <MemoryRouter location="someLocation" context={{}}>
      <NavLink to="someLocation" />
    </MemoryRouter>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <MemoryRouter location="someLocation" context={{}}>
        <NavLink to="someLocation" />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains a `li` subcomponent with a class `navigation-link`', () => {
    expect(component.find('li')).toHaveLength(1);
    expect(component.find('li').hasClass('navigation-link')).toEqual(true);
  });

  it('contains an `Link` subcomponent', () => {
    expect(component.find(Link)).toHaveLength(1);
  });

  it('contains an `a` subcomponent', () => {
    expect(component.find('a')).toHaveLength(1);
  });
});
