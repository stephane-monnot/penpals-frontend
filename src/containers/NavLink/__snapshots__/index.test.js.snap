// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props =>
_react2.default.createElement('li', { __source: { fileName: _jsxFileName, lineNumber: 6 }, __self: undefined },
  _react2.default.createElement(_reactRouterDom.Link, Object.assign({}, props, { __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: undefined }))) renders and matches our snapshot 1`] = `
<li>
  <a
    href="someLocation"
    onClick={[Function]}
  />
</li>
`;
